src/cmd/moby/run_qemu.go

```diff
@@ -73,18 +91,38 @@ func runQemu(args []string) {
 	} else {
 		qemuArgs = append(qemuArgs, "-enable-kvm")
 		qemuArgs = append(qemuArgs, "-machine", "q35,accel=kvm:tcg")
+		if container {
+			dockerArgs = append(dockerArgs, "", "")
+			copy(dockerArgs[5:], dockerArgs[3:])
```

Comment:
this looks odd. You set up dockerArgs above and here you are munging it and it hasn't been touched in between. I don't understand why this is necessary.